onebusaway-gtfs/src/main/java/org/onebusaway/gtfs/model/FareLegRule.java

```diff
@@ -54,28 +46,34 @@ public final class FareLegRule extends IdentityBean<String> {
   @CsvField(name = "distance_type", optional = true)
   private Integer distanceType;
 
-  public String getLegGroupId() {
+  @CsvField(
+    name = "fare_product_id",
+    mapping = DefaultAgencyIdFieldMappingFactory .class
+  )
+  private AgencyAndId fareProductId;
```

Comment:
I know we talked about this, but please remind me again why you cannot put the `FareProduct` instance there?

In the spec there is still only a single fare product per fare leg rule. A _group_ of rules may have several fare product but not the individual instance.

Reply:
There is only a single fare_product_id per fare leg rule, but there can be multiple fare products with the same ID, since the PK is (fare_product_id, fare_media_id), and fare_leg_rules.txt does not have a fare_media_id to differentiate between them.

Here's a snippet from the MBTA's fare_products.txt:
```
fare_product_id,fare_product_name,fare_media_id,amount,currency
prod_cr_zone_1,Commuter Rail Zone 1 one-way fare,cash,6.5,USD
prod_cr_zone_1,Commuter Rail Zone 1 one-way fare,credit_debit,6.5,USD
prod_cr_zone_1,Commuter Rail Zone 1 one-way fare,mticket,6.5,USD
```

A fare leg rule with the fare_product_id == "prod_cr_zone_1" would need to match all three of those.

Reply:
I see now. Despite the name a `fare_product_id` no longer identifies a single fare product.